src/engine/game.ts

```diff
@@ -1,33 +1,39 @@
 'use strict';
 
-import GameInfo from './interfaces/game-info';
 import FrameRate from './models/frame-rate'
 import GameObject from './models/game-object';
 import RenderEngine from './services/render-engine';
-import TimeService from './services/time-service';
+import InputController from './input-controllers/input-controller';
+import GameEvent from './events/event';
+import EventSystem from './services/event-system';
+import ArrayUtils from './util/array-utils'
 
 export default class Game {
-  canvas: HTMLCanvasElement;
-  frameRateOverlay: FrameRate;
-  gameInfo: GameInfo;
-  gameObjects: Array<GameObject>;
-  running: boolean;
+  private canvas: HTMLCanvasElement;
+  private context: CanvasRenderingContext2D;
+  private frameRateOverlay: FrameRate;
+  private gameObjects: Array<GameObject>;
+  private static inputControllers: Array<InputController>;
+  private running: boolean;
+  private currentFrame: number;
 
   constructor(canvas: HTMLCanvasElement) {
     this.canvas = canvas;
+    let twoDContext = canvas.getContext('2d');
+    if (twoDContext == null) {
+      throw new Error('Could not get the context from the canvas');
+    }
+    this.context = twoDContext!;
```

Comment:
It is unnecessary because of the null check. But if I hadn't checked for null it would force set context which does not allow null.

It can be removed.